src/define.js

```diff
@@ -238,6 +238,22 @@ define.property = function(typePrototype, prop, definition, dataInitializers, co
 
 	var type = definition.type;
 
+	//!steal-remove-start
+	if(process.env.NODE_ENV !== 'production') {
+		if(!definition.set && definition.get && definition.get.length === 0 && ( "default" in definition ) ) {
+			canLogDev.warn("can-define-object: default value for property " +
+					canReflect.getName(typePrototype)+"."+ prop +
+					" ignored, as its definition has a zero-argument getter and no setter");
+		}
+
+	if(!definition.set && definition.get && definition.get.length === 0 && ( definition.type ) ) {
```

Comment:
By default all definitions have a `type`... we need to check `definition.type && definition.type !== defaultDefinition.type`

Reply:
copy pasta fixed it